File: app/src/main/java/zone/com/zrefreshlayoutdemo/CircleHeaderActivity.java
package zone.com.zrefreshlayoutdemo;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.ImageView;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.ButterKnife;
import zone.com.zrefreshlayout.ZRefreshLayout;
import zone.com.zrefreshlayoutdemo.header.CircleRefresh;
import zone.com.zrefreshlayoutdemo.resistance.Damping2Head8per;

/**
 * Created by fuzhipeng on 2017/1/11.
 */

public class CircleHeaderActivity extends AppCompatActivity {
    @BindView(R.id.iv)
    ImageView iv;
    @BindView(R.id.refresh)
    ZRefreshLayout refresh;
    @BindView(R.id.tv)
    TextView tv;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.unique_feature);
        ButterKnife.bind(this);

        refresh.setIHeaderView(new CircleRefresh());
//        refresh.setIResistance(new Damping2Head8per());

        tv.setText("包含滚动拦截后,准备就绪后,在滚动到特定位置! 还有下拉位置映射，刷新位置更改");
        refresh.setPullListener(new ZRefreshLayout.PullListener() {
            @Override
            public void refresh(final ZRefreshLayout zRefreshLayout) {
                refresh.postDelayed(new Runnable() {
                    @Override
                    public void run() {
                        zRefreshLayout.refreshComplete();
                    }
                }, 2000);
            }

        }, new ZRefreshLayout.PullStateRestListener() {
            @Override
            public void refreshStateRestComplete(ZRefreshLayout zRefreshLayout) {
                iv.setImageResource(R.drawable.aaaaaaaaaaaab);
            }
        });
    }
}

File: app/src/main/java/zone/com/zrefreshlayoutdemo/CircleHeaderActivity.kt
package zone.com.zrefreshlayoutdemo

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.unique_feature.*
import zone.com.zrefreshlayoutdemo.header.CircleRefresh

/**
 * Created by fuzhipeng on 2017/1/11.
 */

class CircleHeaderActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.unique_feature)
        refresh.setIHeaderView(CircleRefresh())
        //        refresh.setIResistance(new Damping2Head8per());

        tv.text = "包含滚动拦截后,准备就绪后,在滚动到特定位置! 还有下拉位置映射，刷新位置更改"
        refresh.setPullListener({ zRefreshLayout -> refresh.postDelayed({ zRefreshLayout.refreshComplete() }, 2000) }, { iv.setImageResource(R.drawable.aaaaaaaaaaaab) })
    }
}
